apps/web/src/pages/brand/tabs/LayoutEditor.tsx

```diff
@@ -1,38 +1,80 @@
 import { useForm, Controller } from 'react-hook-form';
 import { EmailCustomCodeEditor } from '../../../components/templates/email-editor/EmailCustomCodeEditor';
-import { Center, Grid } from '@mantine/core';
+import { Center, Grid, LoadingOverlay, useMantineTheme } from '@mantine/core';
 import { ArrowLeft } from '../../../design-system/icons';
 import { Button, Checkbox, colors, Input, Text } from '../../../design-system';
 import { useEnvController } from '../../../store/use-env-controller';
-import { successMessage } from '../../../utils/notifications';
+import { errorMessage, successMessage } from '../../../utils/notifications';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { createLayout, getLayoutById } from '../../../api/layouts';
+import { useEffect } from 'react';
 
-export function LayoutEditor({ id, goBack }: { id: string; goBack: () => void }) {
+export function LayoutEditor({
+  id = '',
+  editMode = false,
+  goBack,
+}: {
+  id?: string;
+  editMode?: boolean;
```

Comment:
If I remember correctly if setting defaults in the params make the types not optional anymore. Maybe we should double check it for correctness.

Reply:
What do you mean? It is optional when using the component. Not sure I follow what you mean :)

Reply:
Ignore this. I just double checked and I was wrong. 😓